drracket/drracket/private/unit.rkt

```diff
@@ -3265,9 +3267,31 @@
            #t]
           [else #f]))
       
-      (define/public (open-in-new-tab filename)
-        (create-new-tab filename))
-      
+      (define/public (open-in-new-tab filename #:start-pos [start-pos 0] #:end-pos [end-pos 'same])
+        (create-new-tab filename #:start-pos start-pos #:end-pos end-pos))
+      
+      ;; reopen-closed-tab : -> void
+      ;; Opens previously closed tabs. If no tabs were closed in current session, files from 
+      ;; previous sessions are opened.
+      (define/public reopen-closed-tab
+        (lambda ()
+          (define closed-tabs (preferences:get 'drracket:recently-closed-tabs))
+          (define file-to-open
+            (for/or ([file (in-list closed-tabs)])
+              (define filename (first file))
+              (and (file-exists? filename)
+                   (set! closed-tabs (cdr closed-tabs))
```

Comment:
Does this work properly? Maybe better to remove `file-to-open` from `closed-tabs` after this loop completes? And if any files were skipped because they don't exist anymore, those should probably be removed from the preference too.